A4A: Add Resource Center sections

tiagonoronha opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of https://linear.app/a8c/issue/A4A-1651/implement-resource-center-into-product

<img width="3516" height="3882" alt="screencapture-agencies-localhost-3000-learn-resource-center-2025-12-12-15_21_56 (1)" src="https://github.com/user-attachments/assets/c22020fc-94ce-4579-b61b-b3d82631a9bb" />

<img width="1004" height="737" alt="Screenshot 2025-12-12 at 15 34 17" src="https://github.com/user-attachments/assets/8f1ca4a4-52be-4c47-a2b3-396cedd39ae2" />

## Proposed Changes

* Adds Top Resources, Art of the Deal and Browse All sections
* Adds search and multi-field filtering using DataViews
* Fetches resources  from `/agency/resources` endpoint
* Add modal for video playing with support for YouTube. 

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
* Apply this PR to your sandbox, this is only for testing while we wait for more content to be published: 197963-ghe-Automattic/wpcom
* Sandbox `public-api.wordpress.com`
* Open the live link
* Go to Learn > Resource Center. Confirm it looks like the screenshot
* Click on one of the resources with a "Watch Video" CTA button and confirm the modal is shown and the video loads

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

tiagonoronha committed:
Add Resource Center sections

tiagonoronha committed:
Fetch resources from API

tiagonoronha committed:
Refactor cards and filtering logic

tiagonoronha committed:
Add CTA custom labels

tiagonoronha committed:
Fix responsive layout

tiagonoronha committed:
Add modal for video cards

tiagonoronha committed:
Add support for YouTube videos in the video modal

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/hooks/use-resource-cta-label.ts:
```diff
@@ -0,0 +1,21 @@
+import { useTranslate } from 'i18n-calypso';
```
Can we use `import { __ } from '@wordpress/i18n';` so that it'll be easy when we migrate to MSD?

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/images/jetpack.svg:
We have already added these images as part of "Exclusive offers", can we reuse it?

yashwin commented on client/a8c-for-agencies/sections/learn/format-agency-resources.tsx:
Can we keep this file inside `/sections/learn`

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/browse-all-resources.tsx:
```diff
@@ -0,0 +1,243 @@
+					<Spinner />
+				</div>
+			</>
+		);
```
I think it's better to use loading state from DataViews

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/browse-all-resources.tsx:
```diff
@@ -0,0 +1,243 @@
+				</Spacer>
+				<Text variant="muted">{ __( 'No resources available at this time.' ) }</Text>
+			</>
+		);
```
I think it's better to use the empty state from DataViews

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/resource-section.tsx:
Looks like there is no difference between this component & ArtOfTheDeal. Do you think we can reuse it?

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/art-of-the-deal.tsx:
```diff
@@ -0,0 +1,135 @@
+				<Spacer marginBottom={ 12 } />
+			</>
+		);
+	}
```
Can we show the main page loader for loading? `PagePlaceholder`

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/art-of-the-deal.tsx:
```diff
@@ -0,0 +1,135 @@
+					{ ...( ! isVideo && { href: resource.externalUrl, target: '_blank' } ) }
+					onClick={ handleClick }
+					style={ { marginTop: '24px', alignSelf: 'flex-start' } }
+				>
```
I think we should be using DataViews > Grid view here.

yashwin reviewed (commented):
I tested the changes with the mock API endpoint and most of the things work fine.

I have left some comments. Please let me know your thoughts

Also, I noticed it stays in this UI until the API data is loaded. Maybe the PagePlaceholder will solve the issue.

<img width="1573" height="904" alt="Screenshot 2025-12-18 at 1 52 46 PM" src="https://github.com/user-attachments/assets/d6200ee3-58d6-4a18-b36a-65658bbdb3cd" />

tiagonoronha committed:
Replace i18n-calypso with WordPress i18n in resource center

tiagonoronha committed:
Move format-agency-resources to sections/learn

tiagonoronha committed:
Remove duplicate product logos and use exclusive-offers images

tiagonoronha committed:
Refactor resource sections to use shared component

tiagonoronha committed:
Remove wrapper components and use ResourceSection directly

tiagonoronha committed:
Let DataViews handle empty state in browse-all-resources

tiagonoronha committed:
Add isLoading prop to DataViews in browse-all-resources

tiagonoronha committed:
Use PagePlaceholder for loading state in resource center

tiagonoronha committed:
Refactor resource center to use shared ResourceCard component

tiagonoronha committed:
Merge branch 'trunk' into a4a/resource-center/phase2

tiagonoronha committed:
Merge branch 'trunk' into a4a/resource-center/phase2

tiagonoronha commented:
@yashwin I implemented most of your feedback and also improved the main components further. For example:

* Shared card component used across all three sections instead of separate ones for each.
* Use PagePlaceholder for loading state in resource center
* Removed duplicate product logos
* Replace i18n-calypso with WordPress i18n in resource center

yashwin commented on client/a8c-for-agencies/sections/learn/resource-center/overview-content/resource-card.tsx:
```diff
@@ -0,0 +1,80 @@
+	};
+
+	return (
+		<Card isBorderless={ isBorderless } size={ isBorderless ? 'none' : undefined }>
+			<CardBody style={ { display: 'flex', flexDirection: 'column', height: '100%' } }>
+				<VStack spacing={ 4 } style={ { flex: 1, justifyContent: 'flex-start' } }>
+					{ showPreviewImage && resource.previewImage && (
+						<CardMedia style={ { borderRadius: '4px' } }>
+							<img src={ resource.previewImage } alt={ resource.name } />
+						</CardMedia>
+					) }
+					{ showLogo && <HStack>{ resource.logo }</HStack> }
+					<VStack spacing={ 1 }>
+						<Text size={ 13 } weight={ 500 }>
+							{ resource.name }
+						</Text>
+						<Text variant="muted" size={ 12 }>
+							{ resource.description }
+						</Text>
+					</VStack>
+				</VStack>
+				<Button
+					variant="secondary"
+					{ ...( ! isVideo && { href: resource.externalUrl, target: '_blank' } ) }
+					onClick={ handleClick }
+					style={ { marginTop: '24px', alignSelf: 'flex-start' } }
+				>
+					{ ctaLabel }
+				</Button>
+			</CardBody>
+		</Card>
+	);
```
It's better to replace this with DataViews > Grid instead of creating a new one. What do you think?

yashwin reviewed (approved):
Thanks for addressing my comments. LGTM now 👍 

Let's address my other comment in another PR 👍
